var/spack/repos/builtin/packages/python/package.py

```diff
@@ -303,6 +304,57 @@ def determine_variants(cls, exes, version_str):
             except ProcessError:
                 variants += '~tix'
 
+        # Get config args for debug, and optimizations variant checks
+        sysconf_imp = 'import sysconfig'
+        if version < Version('2.7'):
+            sysconf_imp = 'import distutils.sysconfig as sysconfig'
+        conf_args = python(
+            '-c',
+            '%s; print(sysconfig.get_config_var("CONFIG_ARGS"))' % sysconf_imp,
+            output=str, error=str)
+
+        libdir = python(
+            '-c', '%s; print(sysconfig.get_config_var("LIBPL"))' % sysconf_imp,
```

Comment:
I think we would want to get the directory *containing* `LIBPL`?

Reply:
`os.path.join(sysconfig.get_config_var('LIBPL'), '../..')` is working on my osx and linux machines.

Reply:
No, LIBPL is the directory containing libraries, config headers, etc. It's described in the code comment as "Library and miscellaneous stuff needed for extending/embedding".

Reply:
Ah. Well it does not appear to contain `libpython3.so`, just `libpythonX.Y.a` (on python 2 and 3). Are you also seeing this behavior?

Reply:
buuuuuuuuut, it looks like `gcc -shared -o libpythonX.Y.so libpythonX.Y.a` *does* work, so we could perhaps replace it with that? Do `libs` in spack refer to both static and shared libraries?

Reply:
It looks like `libpythonX.y.a` will work as well to build the clingo python module (just checked)! Wonderful.

Reply:
I think we can also change the `libs()` method:
```python
    @property
    def libs(self):
        return find_libraries('libpython*', root=self.get_config_var('LIBPL'), shared=False)
```